pypeit/par/pypeitpar.py

```diff
@@ -2234,6 +2234,17 @@ def __init__(self, telgridfile=None, sn_clip=None, resln_guess=None, resln_frac_
                                'must be downloaded from the GoogleDrive and installed in your PypeIt installation via ' \
                                'the pypeit_install_telluric script. NOTE: This parameter no longer includes the full ' \
                                'pathname to the Telluric Grid file, but is just the filename of the grid itself.'
+        
+        defaults['tell_npca'] = 5
+        dtypes['tell_npca'] = int
+        descr['tell_npca'] = 'Number of telluric PCA components used. Can be set to any number from 1 to 10.'
+        
+        defaults['teltype'] = 'pca'
+        dtypes['teltype'] = str
```

Comment:
I think you also need to add the following in:

`options['teltype'] = TelluricPar.valid_teltype()`

and then add the following function within this class

```
    @staticmethod
    def valid_teltype():
        """
        Return the valid telluric methods.
        """
        return ['pca', 'grid']
```

Reply:
It looks like a lot of the classes are missing `options`! I have added it, nevertheless, hopefully in the correct way.